python/tank/platform/bundle.py

```diff
@@ -940,7 +957,19 @@ def __resolve_hook_path(self, settings_name, hook_expression):
             hooks_folder = os.path.join(engine.disk_location, "hooks")
             path = hook_expression.replace("{engine}", hooks_folder)
             path = path.replace("/", os.path.sep)
-        
+
+        elif hook_expression.startswith("{show}"):
+            # this is expected to be used only when there are show overrides, and hence
+            # possible declaration of correct hook name in the sgtk_config preferences files.
+            path = hook_expression.replace("{show}/", "")
```

Comment:
This is a little too loosy goosey for me. We should lock the hook path down to a specific location within the level hierarchy. Perhaps `<level>/SHARED/etc/sgtk/hooks` ?

Reply:
Sure, I will add the tighter path in the core hook.